kernel_gateway/services/notebooks/handlers.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+import tornado.web
+from tornado.log import access_log
+import json
+from .request_utils import *
+try:
+    from queue import Empty
+except ImportError:
+    from Queue import Empty
+
+class NotebookAPIHandler(tornado.web.RequestHandler):
+    kernel_client = None
+    sources = None
+    execution_timeout = 5
+    kernel_name = ''
+
+    def initialize(self, sources, kernel_client, kernel_name):
+        self.kernel_client = kernel_client
+        self.sources = sources
+        self.kernel_name = kernel_name
+
+    def _send_code(self, code, block=True):
+        # execute the code and return the result and HTTP status code
+        self.kernel_client.execute(code)
+        result_found = False
+        iopub_message = None
+        result = None
+        status = 200
+        try:
+            while(result is None):
+                iopub_message = self.kernel_client.get_iopub_msg(block=block, timeout=self.execution_timeout)
```

Comment:
For now, block=True is fine. We need to think about if we want to support running multiple kernels behind the gateway and having the gateway act as a load balancer of sorts across them or not. Right now, we're assuming the kernel+gateway combo will be scaled out by a higher power.